### ISSUE TITLE
Classes at a University

### ISSUE THREAD

**Issue by genkimarshall** [genkimarshall/feedback-wanted#2]
These are some *still-in-development* thoughts after TA'ing a semester of `CS 2800 - Discrete Structures`. TA again. I know none of these really are *new* ideas, and as with everything, the real problem is not the lack of ideas but implementing them correctly. But, I would hope to get feedback on, and improve, the below. My feelings on them are definitely not concrete, but I think the perspectives are valid. One of the improvements needed is dividing between fundamental changes and changes that can be done now by students.

### Unsorted
* I wish I could TA every class. I got so much more out of trying to teach Discrete Structures to other students than just learning myself.
* Ideally, *every student* should be able to give tutoring to the next semester's class. If teaching wasn't their thing, it would still be worth the experience for all because of [support groups](http://www.gatesnotes.com/Education/The-Future-of-College-NACUBO-Remarks). It would help with the issue of overflowing office hours where students don't get enough individual attention.
* There needs to be a better system for figuring out which students are struggling with which subjects and holding batch office hours or writing posts accordingly to resolve misunderstandings. Too much time wasted where the class moves on without students fully understanding a building-block subject.
* We, as a group of TA's and professors, continually found better ways to run things throughout the semester. Unfortunately, it felt like these problems should have (and most likely have) been solved before by others at the university. It's unfortunate that these lessons get lost. Not sure what to do here.

### Piazza
* Piazza is amazing. I could go on and on about it, but to avoid side-tracking myself, I'll just say that it is really addictive to answer student's questions on there, and that's a great thing. Students helping other students is the ideal. I think the biggest thing is community/social trust: see [here](http://www.nytimes.com/2014/08/17/opinion/sunday/teaching-is-not-a-business.html?_r=0).
* In-person office hours are helpful and all I think, but there is too much variance and potentially mistakes that TA's make. I personally think something like Piazza is way more ideal, as TA's can take way more time to develop a well-developed answer, TA's and professors can correct each other's answers, and the best solutions can be found, saved, and used repeatedly and easily.

### One class at a time
* I wish I could have my students able to focus on one class at a time.
* I think there is just too much wasted time having to deal with context-switching between classes. I think an ideal situation would be to have students take one class at a time, getitng to spend more time on it, have it each day, working on it with other students (taking the class or who already took the class), and the professors.
* You could start or stop a class at any time with no penalties, which would help encourage people to try things outside their self-perceived ability or skillset. I believe so many benefits come out of that.
* I want a system of moving-at-your-own-pace system so badly, because it was so distressing to see students come in much later saying how they "never really understood" subjects X, Y, Z. Even though they are so fundamental to what came later!
* A TA for the functional programming class that I took this past semester said he/she wanted to reward *fluency* in OCaml. I believe this sort of focused-fluency can only really happen when taking an immersive, every day class system. Like my Japanese class, that has section every day where you put into practice what you learned the previous lecture/night.

### Not classes, units
* I guess it would go along with the above idea, except the idea is 'one unit at a time', where units build on top of each other, and you can change subjects as you wish. [Relevant discussion](https://www.reddit.com/r/IAmA/comments/2rgsan/i_am_elon_musk_ceocto_of_a_rocket_company_ama/cnfvx6u).
* This would minimize what I believe is the worst inhibitor to learning: Doing something that you don't want to be doing. It leads to procrastination, distraction, frustration, ineffective memories, etc etc etc. 
* Outside of TA'ing, the biggest problem I experienced myself and many other students too: Lack of a good system to figure out what you want to study. And this is essential to wanting to do what you are doing. That gets me really worked up, I feel I have a lot to say about it, but moving on.
* It goes along the idea that huge commitments are intimidating and not conducive to figuring out what you want to do. The process of finding out what you want to do should be as friction-less as possible.

### Problem Sets
* Not-required-but-recommended practice problems that students, TA's, and professors can submit and students can do and get social recognition / non-grade-points for doing, review each other's solutions, etc. Looking to expand more on this later.

### Starting this without fundamental changes
* Example: could start by doing expanded versions of my current study group with CS 2800 for each class. It would be a social group that convenes at specified times to do the homeworks, and everyone would come together to work alone, but to ask each other questions when needed.
* These groups would meet in days leading up to the homework assignment, but never the night before, to minimize rampant copying and stress.
* I think a super important requirement of this group is that before attending a session for a homework assignment, you would have to have spent legitimate effort on each problem, understand the nuances, and where you are stuck. And match people up that way (depending on their progress).
* When people finish, they can go help others if they feel like it.
* In each class, there would be a tier system where you would get checked off if you have gotten to a certain point (e.g. completed a certain homework, and proved you know how to do all the questions)
* If you couldn't, you would receive help from members until you understood it and proved it.
* Would give good incentive to people to stay on course, and the incentive to even try for these things is the access you get to other peers working on higher assignments.
* The focus would be advertising this as a social club that learns together. Hang out with people, make new friends, via trying to solve problems and learn, while getting better grades for having done so.

### One last thought
* This is definitely the least-developed, but here goes.
* Perhaps the more general solution to a lack of good teachers, is that we shouldn't need so many good teachers. Maybe what we really just need is students teaching other students, with a minimal amount of teachers to help it along.
* Students know better how to sympathize with other students, they know what it felt like just recently, they get to learn the material better, students will prefer learning from people who they feel understand them better, it encourages caring about school because you're powering through the work with friends, etc.

**Comment by gharrma** [comment]
I fill my Evernote with little notes like these. A couple years ago I wrote: 
"Reason to support more focused education (e.g. block scheduling): greater focus leads to more rapid learning in a single subject. When learning is noticeable to a student, he or she is more likely to be inspired and motivated."

Anyway, I seriously loved reading this. I'm busy now (teaching programming to teens at a tech camp!), but I'm definitely going to come back to this and think about these ideas more. Would love to discuss them :D

**Comment by genkimarshall** [comment]
Wow, I can't express enough how appreciative I am that someone read this, thanks so much :)
Haha that is so funny, I used to use Evernote too (now using something else, but I remember loving Evernote), and also have stuff like this all over my notes. I have come to be sad that they just sit there collecting dust in the dark, so that's why I had to open them.
I would also really love to discuss these more, they are really vague but I really want to discuss them with people. It's hard and time-consuming to know which thoughts to discuss with who though. So another reason I'm glad you found it :D So yeah sounds good!! :+1:

**Comment by chiragbharadwaj** [comment]
I agree that everybody should have the opportunity to TA! It really helps, and being able to build an interactive community through Piazza allows both students and TA's to come up with ingenuous ways of solving problems. Ideally, everybody would get this opportunity, not just the people who study the most/like helping the most.

Helping people without showing them how to do something is a great skill to have, in my opinion! I also really like the idea of recommended but optional problem sets! It's nice to build a collective repository of interesting and insightful problems that go beyond the course material in novel, interesting ways so everybody can benefit mutually (including the professors, perhaps!).

One additional suggestion I have to add to your list is to open up the idea of study groups beyond just the university. These days, it's not at all common to see students from different universities bounce ideas off of each other unless it's for a personal project or hackathon or something. What if we could make intra-national (for now... maybe it could be international one day, once we mandate the same standard of excellence at all learning institutions) Piazza that connected students from different universities and let them study together as well? We could integrate curricula better. One potential downside to this that I can see is that those with less fortunate backgrounds might feel overwhelmed by those with more fortunate backgrounds who happen to know a lot more. But then again, this is an issue that perpetuates even just a single college.

Any thoughts?

**Comment by genkimarshall** [comment]
Hey thanks for responding too :D Sincerely really appreciated.

I have only tried/started a couple MOOCs before but I've realized they're not really my learning style.. But maybe I tried the wrong ones.. But I think that's what some MOOCs have tried to provide for courses, right? Like a standardized course that anybody can take, along with forums that people can discuss problems/solutions together with each other. Maybe that's rare though. I _do_ feel that an ideal end goal would be to have global MOOCs be supported by local teachers/students that help guide students through the MOOCs. That way we can get the standardized course material alongside things like Piazza where students from anywhere could discuss things. And really importantly, the face-to-face help. But the thing is, that can be already done can't it? Anyone can create a Piazza institution and host a 'class' there that would go along with some MOOC that people agree to take together and help other students. So I think the barrier still is the fact that universities in general don't seem eager to try radical things like this... I think... from what I've seen so far.

What you said about mandating the same standard of excellence at all learning institutions... I like that a lot. I think it is an ideal goal that I believe can be a very real goal. A wuote from the Bill Gates piece I linked above that I feel I want to CP here... :D

> The MOOC, by itself, doesn’t really change things, except for the very most motivated student. It’s just an element to be mixed in to get all the steps to get through an entire degree program. And, so, most of these systems will be hybrid systems. After all, a student who could deal with just the MOOC by itself, without any face-to-face contact and counseling, they’re the type of student who, when we had text books, were also capable of getting by and learning the material.  The MOOC is not based on new educational knowledge. It’s simply presented in an easier to understand, more interactive way that can be fantastic. So, that’s an opportunity.
